src/oauth-manager/oauth-utils.ts

```diff
@@ -498,6 +498,41 @@ export interface ParsedApprovalResult {
   }
   
 
+/**
+ * Validates and sanitizes a URL to ensure it's a valid ThoughtSpot instance URL
+ * @param url - The URL to validate and sanitize
+ * @returns The sanitized URL
+ * @throws Error if the URL is invalid
+ */
+function validateAndSanitizeUrl(url: string): string {
+    try {
+        // Remove any whitespace
+        const trimmedUrl = url.trim();
+        
+        // Add https:// if no protocol is specified
+        const urlWithProtocol = trimmedUrl.startsWith('http://') || trimmedUrl.startsWith('https://') 
+            ? trimmedUrl 
+            : `https://${trimmedUrl}`;
+        
+        const parsedUrl = new URL(urlWithProtocol);
+        
+        // Ensure it's using HTTPS
+        if (parsedUrl.protocol !== 'https:') {
+            throw new Error('URL must use HTTPS');
+        }
```

Comment:
On line 513 you allowed `http` and now you are asserting `https` only. We should allow `http` as well in my opinion as nebula dev clusters would be `http`.

Reply:
My bad, removed this